StoreKitUnitTests/PriceFormatterProviderTests.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PriceFormatterProviderTests.swift
+//
+//  Created by Juanpe Catalán on 10/3/22.
+
+import Nimble
+import StoreKit
+import XCTest
+
+@testable import RevenueCat
+
+class PriceFormatterProviderTests: StoreKitConfigTestCase {
+
+    private var priceFormatterProvider: PriceFormatterProvider!
+
+    override func setUp() {
+        super.setUp()
+
+        self.priceFormatterProvider = PriceFormatterProvider()
+    }
+
+    func testReturnsCachedPriceFormatterForSK1() {
+        let locale = Locale(identifier: "en_US")
+        let firstPriceFormatter = priceFormatterProvider.priceFormatterForSK1(withLocale: locale)
+
+        let secondPriceFormatter = priceFormatterProvider.priceFormatterForSK1(withLocale: locale)
+
+        XCTAssertIdentical(firstPriceFormatter, secondPriceFormatter)
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testReturnsCachedPriceFormatterForSK2() throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let currencyCode = "USD"
+        let firstPriceFormatter = priceFormatterProvider.priceFormatterForSK2(withCurrencyCode: currencyCode)
+
+        let secondPriceFormatter = priceFormatterProvider.priceFormatterForSK2(withCurrencyCode: currencyCode)
+
+        XCTAssertIdentical(firstPriceFormatter, secondPriceFormatter)
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testSk2PriceFormatterUsesCurrentStorefront() async throws {
```

Comment:
I only created a test for SK2 because I saw a comment that said:

>`ProductsFetcherSK1` does not detect Storefront changes to invalidate the cache like `ProductsFetcherSK2` does.

Reply:
Yeah, that's true, I forgot about that. The fetcher doesn't detect Storefront changes. We should address that (as a separate thing). 

But independently of that, if the `PriceFormatterProvider` is passed a new locale, it does update the formatter, right? So I think we should test that. 

That way when we do add automatic updates to the fetcher, we'll know that the `PriceFormatterProvider` will respond correctly as long as the test passes

Reply:
Yep, totally agree. We could create a new fetcher after the Storefront has changed and the `PriceFormatterProvider` should respond correctly. Let me change it.

Thanks!